docs/rholang/rholangmatchingtut.md

```diff
@@ -0,0 +1,181 @@
+# A Tutorial for Matching in Rholang
+
+## Patterns we are allowed to match
+
+Let's clarify the implications of the following sentence from the Rholang tutorial: "There are two kinds of values in Rholang, names and processes. Patterns are names or processes with free variables, which appear to the left of an arrow in a `for` or a `match`." This has a couple of implications that were not explicitly stated in the tutorial.
+
+The first is that a program cannot have any free variables. It also can't have any logical connectives `/\` or `\/`, joins, etc unless they form part of a pattern. Logical connectives, joins and arrows can be used if they are in patterns within the program, and any variable in a program must be at most locally free. For example, the following code snippets are not valid programs, despite the fact that they are valid components of patterns:
+
+* `@Nil!(Nil) /\ @Nil!(Nil)`
+* `@Nil!(Nil) ; @Nil!(Nil)`
+* `@Nil!(Nil) , @Nil!(Nil)`
+
+The second, in the same vein, is that a process variable does *not* match with anything that is not a process, meaning that it cannot match with a statement that contains a free variable, a join, a logical connective, etc unless those are written in a pattern fully contained in the statement.  Likewise, a name variable cannot match with anything that is not a quoted process, in the sense that it cannot contain free variables, joins, logical connectives, etc unless they are correctly written in a pattern fully contained in the quoted process. For example, the following code
```

Comment:
In this paragraph you say "it cannot match with ..." and you enumerate multiple things and then you only give two examples. It would be nice to have more examples for each of the situations enumerated above